MakingMusicTool/MakingMusicTool.py

```diff
@@ -0,0 +1,363 @@
+# =========================================================================== #
+
+# =========================================================================== #
+#  SECTION: Imports                                                           
+# =========================================================================== #
+import os
+import csv
+
+from pydub import AudioSegment
+from pydub.playback import play
+# =========================================================================== #
+#  SECTION: Global definitions
+# =========================================================================== #
+# name of the input csv file
+FILENAME = "sound_pattern.csv"
+# =========================================================================== #
+#  SECTION: Class definitions
+# =========================================================================== #
+
+
+class MusicMaker(object):
+    """
+    MusicMaker is a class to that can create soundtracks out of a given pattern.
+    The pattern needs information about a instrument, the volume and a 4/4-time
+    rythm.
+    """
+    # ----------------------------------------------------------------------- #
+    #  SUBSECTION: Constructor
+    # ----------------------------------------------------------------------- #
+
+    def __init__(self, fileName: str, bpm:int=60):
+        ## public
+        # input from the csv file
+        self.sound_pattern = self.analyse_data(fileName)
+        ## __private
+        # given instrument beat audio files (wav-format!!!)
+        self.__instruments =   {'drum': 'mixkit-metal-hit-drum-sound-550.wav',
+                                'bass': 'mixkit-bass-guitar-single-note-2331.wav',
+                                'tribal drum': 'mixkit-tribal-dry-drum-558.wav',
+                                'violin': 'mixkit-orchestral-violin-jingle-2280.wav'}
+        # convert the beats per minutes (bpm) into the duration of one 4/4 time [ms]
+        self.__44time_duration = 60*4/bpm*1000
+        # reference volume for the used audio files
+        self.__ref_dBFS = -25
+        
+
+    # ----------------------------------------------------------------------- #
+    #  SUBSECTION: Getter/Setter
+    # ----------------------------------------------------------------------- #
+
+    # ----------------------------------------------------------------------- #
+    #  SUBSECTION: Public Methods
+    # ----------------------------------------------------------------------- #
+    def analyse_data(self, fileName:str)->dict:
+        """
+        Import of the csv data and converting it into a dict of sound patterns.
+
+        Parameters
+        ----------
+        fileName : str
+            name of the csv data file
+
+        Returns
+        -------
+        dict
+            sound pattern of the csv file
+        """
+        # make absolute path
+        path = os.path.dirname(os.path.abspath(__file__))
+        abs_path = os.path.join(path, fileName)
```

Comment:
i would use the more verbose  word absolute instead of abs
```suggestion
        absolute_path = os.path.join(path, fileName)
```